pandas/tests/series/test_constructors.py

```diff
@@ -156,10 +156,20 @@ def test_constructor_series(self):
 
         assert_series_equal(s2, s1.sort_index())
 
-    def test_constructor_iterator(self):
+    def test_constructor_iterable(self):
+        class Iter():
+            def __iter__(self):
+                for i in range(10):
+                    yield i
 
         expected = Series(list(range(10)), dtype='int64')
-        result = Series(range(10), dtype='int64')
+        result = Series(Iter(), dtype='int64')
```

Comment:
cant this simply be iter(range(10))?

Reply:
`range` is list-like, subclass of `Sequence`, which doesn't need  consuming by `data = list(data)`. I'd like to add back `range(10)` test and keep both.